autoload/codefmt/shfmt.vim

```diff
@@ -0,0 +1,91 @@
+" Copyright 2017 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+let s:plugin = maktaba#plugin#Get('codefmt')
+
+
+""
+" @private
+" Formatter: shfmt
+function! codefmt#shfmt#GetFormatter() abort
+  let l:formatter = {
+      \ 'name': 'shfmt',
+      \ 'setup_instructions': 'Install shfmt and configure the ' .  
+          \ 'shfmt_executable flag'}
+
+  function l:formatter.IsAvailable() abort
+    return executable(s:plugin.Flag('shfmt_executable'))
+  endfunction
+
+  function l:formatter.AppliesToBuffer() abort
+    return &filetype is# 'sh'
+  endfunction
+
+  ""
+  " Reformat the current buffer with shfmt or the binary named in
+  " @flag(shfmt_executable), only targeting the range between {startline} and
+  " {endline}.
+  function l:formatter.FormatRange(startline, endline) abort
+    let l:Shfmt_options = s:plugin.Flag('shfmt_options')
+    if type(l:Shfmt_options) is# type([])
+      let l:shfmt_options = l:Shfmt_options
+    elseif maktaba#value#IsCallable(l:Shfmt_options)
+      let l:shfmt_options = maktaba#function#Call(l:Shfmt_options)
+    else
+      throw maktaba#error#WrongType(
+          \ 'shfmt_options flag must be list or callable. Found %s',
+          \ string(l:Shfmt_options))
+    endif
+    " Hack range formatting by formatting range individually, ignoring context.
```

Comment:
Could you follow up with the tool maintainer to see about adding range formatting?